src/Microsoft.Diagnostics.Runtime/src/Common/ParallelObjectSet.cs

```diff
@@ -18,7 +18,7 @@ public override bool Contains(ulong obj)
             {
                 int offset = GetOffset(obj, seg);
 
-                lock (seg.Objects.SyncRoot)
+                lock (seg.Objects)
```

Comment:
What's the reasoning behind this change?  I don't mind using the object itself instead of a syncroot, but I'm not clear why you made the change here.

Reply:
I see `ICollection.SyncRoot` as essentially deprecated (based on the fact it's only implicitly implemented in newer BCL collections such as `List<T>`.

There's little functional difference in this change, though `BitArray.SyncRoot` does lazily allocate an object to return which this change avoids.